pytorch_lightning/trainer/trainer.py

```diff
@@ -1099,6 +1101,9 @@ def _pre_training_routine(self):
         # register auto-resubmit when on SLURM
         self.slurm_connector.register_slurm_signal_handlers()
 
+        # used to register gracefully detection of signal to ensure clean exit mechanism
+        self.fault_tolerant_connector.register_fault_tolerant_signal_handlers()
```

Comment:
note it's probably also equally important to de-register the handlers on teardown. or to at least provide one so people can unregister them. what do you think?

Reply:
Yes, I will do this in another PR.